### PR TITLE
Fix issue with PID entries in Redis not being deleted for failed jobs

### PR DESCRIPTION
**Pull request by martinbutt**
Replaces https://github.com/resque/php-resque/pull/50

### CODE REVIEW

**Comment by martinbutt** [comment]
And as a note, this fixed the Redis leak and has been working perfectly in a high traffic (100M+ keys per day) production env since it was committed a couple of years ago.

**Comment by pprkut** [comment]
This will "conflict" with #56.

@danhunsaker I think you can merge this one first and then I'll adjust the namespaces pull request for it.

**Review by pprkut — APPROVED** [review]

**Comment by danhunsaker** [comment]
Drat it all. Missed these comments before merging #56...

File: lib/Resque/Job.php [hunk 1]
```diff
@@ -243,6 +243,11 @@ public function fail($exception)
 				$this->queue
 			);
 		}
+		
+		if(!empty($this->payload['id'])) {
+			Resque_Job_PID::del($this->payload['id']);
```
Comment (pprkut) [inline lib/Resque/Job.php:248]: With namespaces this should now be `Resque\Job\PID`. Please add a use statement at the top and use `PID` here.
The filename of this file changed from `Job.php` to `JobHandler.php`

**Comment by danhunsaker** [comment]
@martinbutt - Can we rebase and address the comment(s) on this so we can GTM?

**Comment by martinbutt** [comment]
It was easier to do a new PR. See #74.

**Pull request closed without merging**